src/core/reactor.cc

```diff
@@ -1132,9 +1156,10 @@ void cpu_profiler::on_signal() {
         return;
     }
 
-    // Skip the sample if another task is currently reading
+    // Skip the sample if the main thread is currently reading
     // _traces. This case shouldn't happen often though.
-    if(_traces_mutex.try_lock()) {
+    auto guard_opt = _traces_mutex.try_lock();
+    if(guard_opt.has_value()) {
```

Comment:
do we want:

```
if (auto guard_opt = _traces_mutex.try_lock(); guard_opt.has_value())
```

here to match what the code was before? I think now we also keep the lock while rearming the timer.

Reply:
Good catch, switching over this.

Reply:
Though in this case it doesn't matter much as nothing else can acquire the mutex while the interrupt handler is running.